sonarqube-companion-rest/src/main/java/pl/consdata/ico/sqcompanion/IndexHtmlController.java

```diff
@@ -0,0 +1,76 @@
+package pl.consdata.ico.sqcompanion;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@RestController
+public class IndexHtmlController {
+
+    private final ResourceLoader resourceLoader;
+    private final String indexTemplatePath;
+
+    public IndexHtmlController(
+            final ResourceLoader resourceLoader,
+            final @Value("${app.indexTemplatePath:classpath:/resources/index.html}") String indexTemplatePath) {
+        this.resourceLoader = resourceLoader;
+        this.indexTemplatePath = indexTemplatePath;
+    }
+
+    @RequestMapping({"/", "/index.html"})
+    public String indexHtml(@RequestParam("baseHref") Optional<String> baseHrefParam, @RequestParam Map<String,String> requestParams) {
+        try (final InputStream indexInputStream = getIndexResource(indexTemplatePath).getInputStream()) {
```

Comment:
na tym etapie to dla mnie premature optimization ;-) btw. gdyby podać inny w configu to mógłby się zmienić między wywołaniami, więc jak już to cache.